specification.md

```diff
@@ -1357,44 +1359,86 @@ For more information about workflow expressions, reference the [Workflow Express
+or it can be defined as an [expression function definition](#Using-Functions-For-Expression-Evaluation) 
+and then applied using an _expression reference string_.
+
+An _expression string_ starts with the `${` token and ends on `}`. All text 
+between the curly brackets is evaluated using the workflow's specified 
+[`expressionLang`](#Workflow-Definition-Structure).
+Note, that without the closing bracket at the very end of the string, it won't 
+be evaluated as an expression.
+
+A workflow expression can also be defined in an [expression function definition](#Using-Functions-For-Expression-Evaluation) and then applied by reference.
+To reference a expression function, an _expression reference string_ is used that 
+starts with the `$fn{` token and ends with `}`.
+The _expression reference string_ has the format `$fn{<function name>}`, where 
+`<function name>` has to match the user-specified `name` of the defined 
+expression function of `type` `"expression"`.
+
+On the rare occasion that one needs to create a string that literally looks like 
+an _expression string_ or an _expression reference string_, the dollar sign 
+should be repeated. Only if the remainder matches a workflow expression markup,
+the runtime reduces the double dollar signs `$$` to a single dollar sign `$` and 
+refrains from evaluating it as a workflow expression.
 
-Note that using a non-default expression language could lower the portability of your workflow definitions 
-across multiple container/cloud platforms.
+```text
+"foo": "$${ this is text in a string starting with '${' and ending with a single '}' }"
+"foo": "$$fn{ this is text in a string starting with '$fn{' and ending with a single '}' }"
+```
+
+The expression language gives access to the state data to select or transform 
+the state data when processing the workflow expression. It also provides access 
+to [workflow secrets](#Workflow-secrets) and 
+[workflow constants](#Workflow-constants).
+
+Note that different data filters play a big role as to which parts of the 
+states data are to be used when the expression is evaluated during workflow
+execution, e.g. a [State Data Filter](#State-data-filters) `input`
+expression can only access data that has been passed to that state, inside an action, 
+only the data filtered by the [Action Data Filter](#Action-data-filters) `fromStateData` 
+parameter are available to generate arguments to an action, etc.
+Please refer to the specification of filters for more inforamtion, 
+e.g. [State Data Filter](#State-data-filters), [Action Data Filter](#Action-data-filters), 
+[Event Data Filter](#Event-data-filters) and see the specification of individual 
+parameters if you want to set them using an expression.
+
+The default expression language is [jq](https://stedolan.github.io/jq/).
+This version of the workflow language supports jq [version 1.6](https://github.com/stedolan/jq/releases/tag/jq-1.6) syntax.
```

Comment:
Yes, it's meant to be jq 1.6. Previous versions of jq have issues and I think it is valid for a spec to fix it to a specific version. Evaluating what jq adds in future versions, we can assess if we want to update this to be only a minimum requirement.

Reply:
What would such a spec change look like, and how would a workflow specify jq version 1.6 after the spec updates to allow support for later versions? I think it would be more accurate to update these lines to something like
> The default expression language is [jq version 1.6](https://github.com/stedolan/jq/releases/tag/jq-1.6). A future revision may update this default, requiring workflows that require continued use of that exact language to specify it explicitly.